src/Module/Contact.php

```diff
@@ -355,54 +358,45 @@ public static function content(array $parameters = [], $update = 0)
 				throw new NotFoundException(DI::l10n()->t('Contact not found'));
 			}
 
-			if ($cmd === 'update' && ($orig_record['uid'] != 0)) {
-				self::updateContactFromPoll($contact_id);
-				DI::baseUrl()->redirect('contact/' . $contact_id);
+			$cdata = Model\Contact::getPublicAndUserContactID($orig_record['id'], local_user());
```

Comment:
You haven't checked for an empty `$cdata` here. Should or shouldn't we handle this?

Reply:
We should check the output of most methods, you're right.